api/restHandler/GlobalPluginRestHandler.go

```diff
@@ -37,6 +45,96 @@ type GlobalPluginRestHandlerImpl struct {
 	enforcerUtil        rbac.EnforcerUtil
 	enforcer            casbin.Enforcer
 	pipelineBuilder     pipeline.PipelineBuilder
+	userService         user.UserService
+}
+
+func (handler *GlobalPluginRestHandlerImpl) PatchPlugin(w http.ResponseWriter, r *http.Request) {
+	decoder := json.NewDecoder(r.Body)
+	userId, err := handler.userService.GetLoggedInUser(r)
+	if userId == 0 || err != nil {
+		common.WriteJsonResp(w, err, "Unauthorized User", http.StatusUnauthorized)
+		return
+	}
+	var pluginDataDto plugin.PluginMetadataDto
+	err = decoder.Decode(&pluginDataDto)
+	if err != nil {
+		handler.logger.Errorw("request err, PatchPlugin", "error", err, "payload", pluginDataDto)
+		common.WriteJsonResp(w, err, nil, http.StatusBadRequest)
+		return
+	}
+	handler.logger.Infow("request payload received for patching plugins")
+	// RBAC enforcer applying
+	isSuperAdmin, err := handler.userService.IsSuperAdmin(int(userId))
```

Comment:
we can use casbin for super admin rbac

Reply:
internally it uses casbin to check for super admin